From 44f75ef41659699f5e650c5dab0e1f87e0bbfe13 Mon Sep 17 00:00:00 2001
Message-Id: <44f75ef41659699f5e650c5dab0e1f87e0bbfe13.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: vkorjani <vikas.korjani@intel.com>
Date: Fri, 9 Jan 2015 12:15:38 +0530
Subject: [PATCH 29/29] FOR_UPSTREAM [VPG]: drm/i915: Expand generic vbt i2c
 support

Exapanding generic vbt i2c to support 2 byte register offset
i2c message size should be Payload + 1 and clean up in
misleading error even on succss.

Issue: GMINL-5295
Change-Id: I1267d74123ec89248e1ce307f15e15b84f301c12
Signed-off-by: vkorjani <vikas.korjani@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi_panel_vbt.c |   13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
index d63cb60..afe19bc 100644
--- a/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
+++ b/drivers/gpu/drm/i915/intel_dsi_panel_vbt.c
@@ -485,22 +485,23 @@ static u8 *mipi_exec_i2c(struct intel_dsi *intel_dsi, u8 *data)
 
 	msg.addr   = slave_add;
 	msg.flags  = 0;
-	msg.len    = 2;
+	msg.len    = payload_size + 1;
 	msg.buf    = &transmit_buffer[0];
 
 	do {
 		ret =  i2c_transfer(adapter, &msg, 1);
-		if (ret == -EAGAIN)
+		if (ret == 1)
+			break;
+		else if (ret == -EAGAIN)
 			usleep_range(1000, 2500);
-		else if (ret != 1) {
-			DRM_ERROR("i2c transfer failed %d\n", ret);
+		else {
+			DRM_ERROR("i2c transfer failed, error code:%d", ret);
 			break;
 		}
 	} while (retries--);
 
 	if (retries == 0)
-		DRM_ERROR("i2c transfer failed");
-
+		DRM_ERROR("i2c transfer failed, error code:%d", ret);
 out:
 	kfree(transmit_buffer);
 
-- 
1.7.9.5

